src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -4958,22 +4980,32 @@ static QVariant executeGeomOverlay( const QVariantList &values, const QgsExpress
   FEAT_FROM_CONTEXT( context, feat )
   const QgsGeometry geometry = feat.geometry();
 
-  return overlayFunction( subExpression, subContext, spatialIndex, cachedTarget, geometry, testOnly );
+  return overlayFunction( subExpression, subContext, spatialIndex, cachedTarget, geometry, testOnly, invert, neighbors, max_distance);
 }
 
 // Intersect functions:
 
 typedef bool ( QgsGeometry::*t_relationFunction )( const QgsGeometry &geometry ) const;
 
 template <t_relationFunction T>
-static QVariant intersectFilteredOverlay( QgsExpression &subExp, QgsExpressionContext &subContext, const QgsSpatialIndex &spatialIndex, std::shared_ptr<QgsVectorLayer> cachedTarget, const QgsGeometry &geometry, bool testOnly )
+static QVariant indexedFilteredOverlay( QgsExpression &subExp, QgsExpressionContext &subContext, const QgsSpatialIndex &spatialIndex, std::shared_ptr<QgsVectorLayer> cachedTarget, const QgsGeometry &geometry, bool testOnly, bool invert, int neighbors, double max_distance )
 {
-  const QList<QgsFeatureId> targetFeatureIds = spatialIndex.intersects( geometry.boundingBox() );
+  QgsRectangle intDomain = geometry.boundingBox();
+  intDomain.grow(0.001); //include touches geom
```

Comment:
Maybe "bboxGrow" could be an additional parameter to this method, so it's only done when required and 0 otherwise?

Reply:
I don't know. It can be easily done, but I'm not sure of its pratical utility. Anyway it seems to me a bit hard to explain in docs and helps, and could be confusing for the user.

Reply:
That's only an internal implementation detail, nothing to expose publicly.
The only effect it has is a slightly improved performance (because of a more efficient usage of the spatial index for functions which don't need to grow the bbox).

Reply:
Okkk. Now I see what You mean. A new optional parameter in https://github.com/enricofer/QGIS/blob/8c4a04c7153001b6a89419999acea17e4372a038/src/core/expression/qgsexpressionfunction.cpp#L4887 to be used internally by touches and equals

Reply:
Yep, exactly :+1: